src/hooks/js/useDeviceOrientation.js

```diff
@@ -0,0 +1,27 @@
+import { useState, useEffect } from 'react';
+
+export const useDeviceOrientation = () => {
+  const [orientation, setOrientation] = useState({
+    angle: 0,
+    type: 'landscape-primary'
+  });
+
+  useEffect(() => {
```

Comment:
I suggest using useSyncExternalStore instead of useEffect for a more robust implementation. Here's the [documentation](https://react.dev/reference/react/useSyncExternalStore).

Reply:
Thanks for the comment, that hook is new for me 😁